sns_sqs_communicator/queue/protocol.py

```diff
@@ -0,0 +1,58 @@
+import collections.abc
+import typing
+
+import mypy_boto3_sqs.type_defs
+
+from .. import clients, fifo_attributes_creator
+
+
+class QueueProtocol(typing.Protocol):
+    """Protocol for queue."""
+
+    client: clients.SQSClient
```

Comment:
Client also could be different from sqs if we will use another queue, so `typing.Any` is preferrable here

Reply:
But the whole package idea is communication between sqs and sns

Reply:
Maybe we can leave a space for other implementations? It's a bit weird from my perspective, we move this to separate package and maybe we will use it in future in other projects that may require another queue/topic implementation, but we add only sqs/sns logic and make all protocols useless.
It already supports different implementations (if we will take it from calendar service for example), but we're just shrinking it